src/bundle/Resources/public/js/scripts/toggle.button.state.toggle.js

```diff
@@ -0,0 +1,44 @@
+(function(global, doc) {
+    const toggleFields = doc.querySelectorAll('.ibexa-toggle');
+    const toggleInputs = doc.querySelectorAll('.ibexa-toggle input');
+    const toggleState = (event) => {
+        event.preventDefault();
+
+        if (event.currentTarget.classList.contains('ibexa-toggle--is-disabled')) {
+            return;
+        }
+
+        event.currentTarget.classList.toggle('ibexa-toggle--is-checked');
+
+        const isChecked = event.currentTarget.classList.contains('ibexa-toggle--is-checked');
+        const valueToSet = isChecked ? 1 : 0;
+
+        if (event.currentTarget.classList.contains('ibexa-toggle--radio')) {
+            event.currentTarget.querySelector(`.form-check input[value="${valueToSet}"]`).checked = true;
+        } else {
+            event.currentTarget.querySelector('.ibexa-toggle__input').checked = isChecked;
+        }
+    };
+    const addFocus = (event) => {
+        event.preventDefault();
+
+        if (event.currentTarget.classList.contains('ibexa-toggle--is-disabled')) {
```

Comment:
I think its good to assign conditions to some variables to give them meaning. You can read code faster. :)
```suggestion
        const isTogglerDisabled = event.currentTarget.classList.contains('ibexa-toggle--is-disabled');

        if (isTogglerDisabled) {
```